### ISSUE TITLE
Glob fails with ENOTSUP on Windows when working of network drive

### ISSUE THREAD

**Issue by jvoccia** [isaacs/node-glob#205]
Version 3.2.11: 
Using Grunt which still uses this older version -- If this belongs over there, I can certainly move it. Not sure if you all are still maintaining the older versions.

When doing globs with paths that contain ```"**"``` like:
    ```"src/**/*.js"```` 
The error: 
ENOTSUP, operation not supported on socket 'W:\folder\project\src\file.js' 
Happens and the glob fails, when working off some Windows Network Mapped Drives using DFS (I am not sure of the specific underlying network filesystem details).  

See also:
http://stackoverflow.com/questions/30199739/enotup-using-grunt

It looks like the error occurs specifically when the globber calls:
readdirsync on a file, and then gets the "unsupported" error back.

When it handles the error in glob.js ~Line 606:
```javascript
 // now handle errors, and cache the information
if (er) switch (er.code) {
      case "ENOTDIR": // totally normal. means it *does* exist.
        this.statCache[f] = 1
        return cb.call(this, er)
      case "ENOENT": // not terribly unusual
      case "ELOOP":
      case "ENAMETOOLONG":
      case "UNKNOWN":
        this.statCache[f] = false
        return cb.call(this, er)
      default: // some unusual error.  Treat as failure.
        this.statCache[f] = false
        if (this.strict) this.emit("error", er)
        if (!this.silent) console.error("glob error", er)
          return cb.call(this, er)
    }
  }
```
It does not allows for the error "ENOTSUP" only the error "ENOTDIR", adding 
```case "ENOTSUP": ```
Right after ```case "ENOTDIR": ````
Seems to resolve the problem, but it is hard to know if this is the only situation this error would come up.

**Comment by isaacs** [comment]
Yeah, I wonder if there are any other cases where `ENOTSUP` might be raised by `readdir` other than "this is a regular file".

One approach might be to stat the file when we get an `ENOTSUP` just to make sure it is a real file that exists?

**Comment by isaacs** [comment]
Meh. Adding stat is a pita.  Just gonna treat ENOTSUP the same as ENOTDIR.  It already was shown on a previous `readdir` call, so it's probably there.

**Comment by jvoccia** [comment]
Hey thanks for the quick response.  I can just keep back porting the change until grunt catches up (eventually).

**Comment by reyou** [comment]
I am having this issue while pushing on Azure. Is there any update on this?

** Grunt :: An error occurred. **

------------------------------------------------------------------------
[31mAborted due to warnings.[39m
Running "clean:dev" (clean) task[24m
[33mWarning: ENOTSUP: operation not supported on socket, scandir 'D:\home\site\wwwroot\.tmp\public\concat\production.css' [39m
------------------------------------------------------------------------
Looks like a Grunt error occurred--
Please fix it, then **restart Sails** to continue running tasks (e.g. watching for changes in assets)
Or if you're stuck, check out the troubleshooting tips below.
[4mTroubleshooting tips:[24m

 *-> Are "grunt" and related grunt task modules installed locally?  Run `npm install` if you're not sure.

 *-> You might have a malformed LESS, SASS, CoffeeScript file, etc.

 *-> Or maybe you don't have permissions to access the `.tmp` directory?
     e.g., `D:\home\site\wwwroot\.tmp` ?

     If you think this might be the case, try running:
     sudo chown -R YOUR_COMPUTER_USER_NAME D:\home\site\wwwroot\.tmp
Error: The hook `grunt` is taking too long to load.
Make sure it is triggering its `initialize()` callback, or else set `sails.config.grunt._hookTimeout to a higher value (currently 90000)
    at tooLong [as _onTimeout] (D:\home\site\wwwroot\node_modules\sails\lib\app\private\loadHooks.js:85:21)
    at Timer.listOnTimeout (timers.js:92:15) { [Error: The hook `grunt` is taking too long to load.
Make sure it is triggering its `initialize()` callback, or else set `sails.config.grunt._hookTimeout to a higher value (currently 90000)] code: 'E_HOOK_TIMEOUT' }

**Comment by capaj** [comment]
I had this when Azure was deploying my node.js app. NPM install triggered these errors:

```
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1.
Selected node.js version 5.9.1. Use package.json file to choose a different version.
Selected npm version 3.7.3
Updating iisnode.yml at D:\home\site\wwwroot\bin\iisnode.yml
Installing Node 4.0 RC1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  4 12.3M    4  519k    0     0   437k      0  0:00:28  0:00:01  0:00:27  437k
 20 12.3M   20 2563k    0     0  1180k      0  0:00:10  0:00:02  0:00:08 1180k
 36 12.3M   36 4621k    0     0  1456k      0  0:00:08  0:00:03  0:00:05 1456k
 57 12.3M   57 7307k    0     0  1751k      0  0:00:07  0:00:04  0:00:03 1751k
 71 12.3M   71 9019k    0     0  1738k      0  0:00:07  0:00:05  0:00:02 1798k
 83 12.3M   83 10.3M    0     0  1715k      0  0:00:07  0:00:06  0:00:01 2019k
 98 12.3M   98 12.1M    0     0  1737k      0  0:00:07  0:00:07 --:--:-- 1979k
100 12.3M  100 12.3M    0     0  1751k      0  0:00:07  0:00:07 --:--:-- 1979k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   987    0   987    0     0   1170      0 --:--:-- --:--:-- --:--:--  1170
100  609k    0  609k    0     0   519k      0 --:--:--  0:00:01 --:--:--  519k
npm WARN package.json fs@0.0.2 fs is also the name of a node core module.
npm WARN package.json fs@0.0.2 No description
npm WARN package.json fs@0.0.2 No repository field.
npm WARN package.json fs@0.0.2 No README data
npm WARN package.json https@1.0.0 https is also the name of a node core module.
npm WARN package.json https@1.0.0 No repository field.
npm WARN package.json https@1.0.0 No README data
npm WARN package.json querystring@0.2.0 querystring is also the name of a node core module.
npm WARN package.json stream@0.0.2 stream is also the name of a node core module.
npm WARN package.json util@0.10.3 util is also the name of a node core module.
npm WARN package.json url@0.10.3 url is also the name of a node core module.
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\man\man1\sshpk-sign.1']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\man\\man1\\sshpk-sign.1' }
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\man\man1\sshpk-conv.1']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\man\\man1\\sshpk-conv.1' }
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\man\man1\sshpk-verify.1']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\man\\man1\\sshpk-verify.1' }
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\bin\sshpk-conv']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\bin\\sshpk-conv' }
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\bin\sshpk-verify']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\bin\\sshpk-verify' }
glob error { [Error: ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\bin\sshpk-sign']
  errno: -4049,
  code: 'ENOTSUP',
  syscall: 'scandir',
  path: 'd:\\home\\site\\wwwroot\\node_modules\\azure-storage\\node_modules\\request\\node_modules\\http-signature\\node_modules\\sshpk\\bin\\sshpk-sign' }
npm ERR! nospc ENOTSUP: operation not supported on socket, scandir 'd:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\man\man1\sshpk-sign.1'
npm ERR! nospc This is most likely not a problem with npm itself
npm ERR! nospc and is related to insufficient space on your system.

npm ERR! System Windows_NT 6.2.9200
npm ERR! command "d:\\Program Files (x86)\\nodejs\\5.9.1\\node.exe" "D:\\Program Files (x86)\\npm\\1.4.28\\node_modules\\npm\\bin\\npm-cli.js" "install" "--production"
npm ERR! cwd d:\home\site\wwwroot
npm ERR! node -v v5.9.1
npm ERR! npm -v 1.4.28
npm ERR! path d:\home\site\wwwroot\node_modules\azure-storage\node_modules\request\node_modules\http-signature\node_modules\sshpk\man\man1\sshpk-sign.1
npm ERR! syscall scandir
npm ERR! code ENOTSUP
npm ERR! errno -4049
npm ERR! not ok code 0
```

**Comment by Moonlighter** [comment]
hi,

Yesterday i try to install angularjs-cli on SharFolder between a host (Windows10) and à VM on VMware 12(Workstation, with Ubuntu 14.04 LTS)

[In my Ubuntu.vmx]

sharedFolder0.present = "TRUE"
sharedFolder0.enabled = "TRUE"
sharedFolder0.readAccess = "TRUE"
sharedFolder0.writeAccess = "TRUE"
sharedFolder0.hostPath = "C:\wamp\www"
sharedFolder0.guestName = "www"
sharedFolder0.expiration = "session"
sharedFolder0.followSymlinks = "TRUE"

I access to my ShareFolder on my VM here /mnt/hgfs/www/

So after i try to install i receive this issues:

EACCES: permission denied, symlink '../semver/bin/semver' -> '/mnt/hgfs/www/ang20/my-demo/node_modules/agent-base/node_modules/.bin/semver'
Error: EACCES: permission denied, symlink '../semver/bin/semver' -> '/mnt/hgfs/www/ang20/my-demo/node_modules/agent-base/node_modules/.bin/semver'
    at Error (native)

Somebody have an idea? Thank's!
